From 857d311771606ad332264719588af016e5320ede Mon Sep 17 00:00:00 2001
Message-Id: <857d311771606ad332264719588af016e5320ede.1645598116.git.github@sicherha.de>
From: Christoph Erhardt <github@sicherha.de>
Date: Wed, 23 Feb 2022 07:34:21 +0100
Subject: [PATCH] Fix dynamic unit test on i686

Signed-off-by: Christoph Erhardt <github@sicherha.de>
---
 test/elf/dynamic.sh | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/test/elf/dynamic.sh b/test/elf/dynamic.sh
index 0ea7ae82..8f9a6628 100755
--- a/test/elf/dynamic.sh
+++ b/test/elf/dynamic.sh
@@ -30,6 +30,14 @@ EOF
 
 $CC -B. -o $t/exe -pie $t/b.o
 count=$(readelf -W --relocs $t/exe | grep -E 'R_[a-zA-Z0-9_]+_RELATIVE' | wc -l)
-readelf -W --dynamic $t/exe | grep -q "RELACOUNT.*\b$count\b"
+
+case "$(uname -m)" in
+i?86)
+  readelf -W --dynamic $t/exe | grep -q "RELCOUNT.*\b$count\b"
+  ;;
+*)
+  readelf -W --dynamic $t/exe | grep -q "RELACOUNT.*\b$count\b"
+  ;;
+esac
 
 echo OK
-- 
2.35.1

